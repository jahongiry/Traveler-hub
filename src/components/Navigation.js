import React from 'react';
import { NavLink } from 'react-router-dom';
import classes from './Navigation.module.css';
import logo from '../img/planet.png';

const Navigation = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/missions',
      text: 'Missions',
    },
    {
      id: 3,
      path: '/myprofile',
      text: 'My profile',
    },
  ];

  return (
    <nav className={classes.navBar}>
      <div className={classes.mainHeader}>
        <img alt="website logo" src={logo} />
        <h1>Space Travaler&apos;s Hub</h1>
      </div>
      <ul>
        {links.map((link) => (
          <li key={link.id}>
            <NavLink to={link.path}>{link.text}</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Navigation;
