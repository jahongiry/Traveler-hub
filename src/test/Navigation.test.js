import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navigation from '../components/Navigation';

it('Builds the snapchot of the navbar component safely', () => {
  const three = renderer
    .create(
      <Router>
        <Navigation />
      </Router>,
    )
    .toJSON();
  expect(three).toMatchSnapshot();
});
