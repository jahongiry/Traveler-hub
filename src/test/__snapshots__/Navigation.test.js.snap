// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Builds the snapchot of the navbar component safely 1`] = `
<nav
  className="navBar"
>
  <div
    className="mainHeader"
  >
    <img
      alt="website logo"
      src="planet.png"
    />
    <h1>
      Space Travaler's Hub
    </h1>
  </div>
  <ul>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className=""
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className=""
        href="/myprofile"
        onClick={[Function]}
      >
        My profile
      </a>
    </li>
  </ul>
</nav>
`;
